libs/structwalk/walktype_test.go

```diff
@@ -247,3 +251,30 @@ func TestWalkTypeVisited(t *testing.T) {
 		".D",
 	}, visited)
 }
+
+func TestWalkSkip(t *testing.T) {
+	type Outer struct {
+		A int
+		B int
+
+		Inner struct {
+			C int
+		}
+
+		D int
+	}
+
+	var seen []string
+	err := WalkType(reflect.TypeOf(Outer{}), func(path *structpath.PathNode, typ reflect.Type) (continueWalk bool) {
+		if path == nil {
+			return true
+		}
+		if path.String() == ".Inner" {
+			return false
+		}
+		seen = append(seen, path.String())
```

Comment:
nit: I'd move append up, to be the first statement, to show that .Inner itself is visited but .Inner.C is not.